x-pack/plugins/security_solution/public/timelines/components/flyout/index.tsx

```diff
@@ -37,6 +37,25 @@ const FlyoutComponent: React.FC<OwnProps> = ({ timelineId, onAppLeave }) => {
     getTimelineShowStatus(state, timelineId)
   );
 
+  const [focusOwnership, setFocusOwnership] = useState(true);
+  const searchListener = useRef(new AbortController());
+
+  const onOutsideClick = useCallback((event) => {
+    setFocusOwnership(false);
+    const classes = event.target.classList;
+    if (classes.contains('euiFieldSearch')) {
+      window.setTimeout(() => event.target.focus(), 0);
+      searchListener.current = new AbortController();
+      event.target.addEventListener(
+        'blur',
+        () => {
+          setFocusOwnership(true);
+        },
+        { signal: searchListener.current.signal }
```

Comment:
So I saw this, and was like cool! I didn't know eventListeners supported `AbortController`, but even in the demo here: https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#add_a_abortable_listener it looks like it doesn't work (once it changes to 3, you're not supposed to be able to change it back). Unless I'm missing something, which I very well could be